### PR TITLE
[1394][1388] Add next cycle page, link up cycles and courses table

### PR DESCRIPTION
**Pull request by fofr**
### Context

- Connect up the work done in #432 and #433 

### Changes proposed in this pull request

- Refactor recruitment cycle into a model as suggested in code review for #432
- Initialise new model in controllers and remove helper
- Add breadcrumbs
- Remove uses of `rollover?` in favour of `provider.rolled_over?`
- Make cycle page cycle aware using RecruitmentCycle model
- Add guidance to explain cycles
- Fix cycle section text on org page
- Add a feature spec and site prism page for recruitment cycles
- Change the default redirect for cycle pages when not in rollover to point to an organisation not the courses page

### CODE REVIEW

File: app/views/courses/details.html.erb [hunk 1]
```diff
@@ -11,6 +11,13 @@
       <li class="govuk-breadcrumbs__list-item">
         <%= link_to @provider.provider_name, provider_path(code: @provider.provider_code), class: "govuk-breadcrumbs__link" %>
       </li>
+      <% if @provider.rolled_over? then %>
```
Comment (dankmitchell) [inline app/views/courses/details.html.erb]: Not sure we need `then`

File: app/views/courses/index.html.erb [hunk 2]
```diff
@@ -11,6 +11,13 @@
       <li class="govuk-breadcrumbs__list-item">
         <%= link_to @provider[:provider_name], provider_path(code: @provider[:provider_code]), class: "govuk-breadcrumbs__link" %>
       </li>
+      <% if @provider.rolled_over? then %>
```
Comment (dankmitchell) [inline app/views/courses/index.html.erb]: Not sure we need `then`

File: app/views/courses/show.html.erb [hunk 3]
```diff
@@ -11,6 +11,13 @@
       <li class="govuk-breadcrumbs__list-item">
         <%= link_to @provider.provider_name, provider_path(code: @provider.provider_code), class: "govuk-breadcrumbs__link" %>
       </li>
+      <% if @provider.rolled_over? then %>
```
Comment (dankmitchell) [inline app/views/courses/show.html.erb]: Not sure we need `then`

File: app/views/providers/show.html.erb [hunk 4]
```diff
@@ -20,12 +20,28 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
     <% if @provider.rolled_over? %>
-      <h2 class="govuk-heading-m"><%= link_to "Current cycle (2019 – 2020)", provider_recruitment_cycle_path(@provider.provider_code, '2019'), data: { qa: 'provider__courses__current_cycle' }, class: "govuk-link" %></h2>
+      <h2 class="govuk-heading-m">
+        <%= govuk_link_to @current_recruitment_cycle.title,
```
Comment (dankmitchell) [inline app/views/providers/show.html.erb]: No need to use `govuk_link_to` here, it just adds another level of abstraction IMO

File: app/views/providers/show.html.erb [hunk 5]
```diff
@@ -20,12 +20,28 @@
...
       <ul class="govuk-list govuk-list--bullet govuk-!-margin-bottom-8">
-        <li>write about your organisation</li>
-        <li>set your contact details</li>
-        <li>publish this information on all course pages</li>
+        <li>manage courses in the current cycle</li>
+        <li>edit locations and vacancies in the current cycle</li>
+      </ul>
+
+      <h2 class="govuk-heading-m">
+        <%= govuk_link_to @next_recruitment_cycle.title,
```
Comment (dankmitchell) [inline app/views/providers/show.html.erb]: No need to use `govuk_link_to` here, it just adds another level of abstraction IMO

File: app/views/recruitment_cycles/show.html.erb [hunk 6]
```diff
@@ -11,14 +11,30 @@
       <li class="govuk-breadcrumbs__list-item" aria-current="page">
```
Comment (dankmitchell) [inline app/views/recruitment_cycles/show.html.erb]: Think we need to remove `aria-current="page"` from here

File: app/views/recruitment_cycles/show.html.erb [hunk 7]
```diff
@@ -11,14 +11,30 @@
...
 
-<h1 class="govuk-heading-xl">Current cycle (2019 – 2020)</h1>
+<h1 class="govuk-heading-xl">
+  <%= @recruitment_cycle.title %>
+</h1>
 
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
+    <% if @recruitment_cycle.next? %>
+      <p class="govuk-body-l">
```
Comment (dankmitchell) [inline app/views/recruitment_cycles/show.html.erb]: Could we move the `<p>` outside of the if/else to make this DRYer?

**Comment by fofr** [comment]
[[S] Courses Table - Next cycle](https://trello.com/c/4gdcwkIg/1394-s-courses-table-next-cycle)

**Comment by fofr** [comment]
[[S] View next cycle - cycle selection page for users](https://trello.com/c/nwMnSpW9/1388-s-view-next-cycle-cycle-selection-page-for-users)

File: spec/requests/recruitment_cycles_spec.rb [hunk 8]
```diff
@@ -1,30 +1,42 @@
...
         get("/organisations/#{provider.provider_code}/2019")
         expect(response.body).to include('Current cycle (2019 – 2020)')
       end
     end
   end
+
+  describe "when visiting a cycle year that doesn’t exist" do
+    scenario 'it 404s' do
+      get("/organisations/#{provider.provider_code}/1999")
+      expect(response.code).to eq("404")
```
Comment (benilovj) [inline spec/requests/recruitment_cycles_spec.rb]: Would the following work in this context?

```suggestion
      expect(response).to have_http_status(:not_found)
```

**Pull request merged**